### PR TITLE
Get earliest version by Catalog Client.

### PR DESCRIPTION
**Pull request by iriza7**
* Get earliest version by Catalog Client.

* Add Unit tests.

Relates-To: OLPEDGE-918.

Signed-off-by: Bautista, Iryna <ext-iryna.bautista@here.com>

### CODE REVIEW

File: @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts [hunk 1]
```diff
@@ -81,6 +81,33 @@ export class CatalogClient {
...
+        const builder = await this.getRequestBuilder(
+            "metadata",
+            HRN.fromString(this.hrn),
+            abortSignal
+        ).catch(error => Promise.reject(error));
+
+        const earliestVersion = await MetadataApi.minimumVersion(builder, {
+            billingTag: request.getBillingTag()
+        }).catch(err =>
+            Promise.reject(`Error getting earliest catalog version: ${err}`)
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts]: Please add test for this case

File: @here/olp-sdk-dataservice-read/test/unit/CatalogClient.test.ts [hunk 2]
```diff
@@ -132,6 +134,32 @@ describe("CatalogClient", () => {
...
+        getEarliestVersionsStub.callsFake(
+            (
+                builder: any,
+                params: any
+            ): Promise<MetadataApi.VersionResponse> => {
+                return Promise.resolve(mockedEarliestVersion);
+            }
+        );
+
+        const catalogRequest = new dataServiceRead.CatalogVersionRequest().withBillingTag(
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/test/unit/CatalogClient.test.ts]: Please do not use the real class in the unit test, just create mocked or stubbed instance of CatalogVersionRequest and use it

File: @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts (hunk 1 above)
Comment (iriza7) [reply @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts]: Added

File: @here/olp-sdk-dataservice-read/test/unit/CatalogClient.test.ts (hunk 2 above)
Comment (iriza7) [reply @here/olp-sdk-dataservice-read/test/unit/CatalogClient.test.ts]: Updated

**Pull request merged**